tests/Application/Messenger/Handler/StripePayoutHandlerTest.php

```diff
@@ -278,18 +278,75 @@ public function testSuccessfulUpdateForRetriedPayout(): void
         $this->assertEquals(SalesforceWriteProxy::PUSH_STATUS_PENDING_UPDATE, $donation->getSalesforcePushStatus());
     }
 
+    /**
+     * This scenario is especially pertinent in March 2024 while we're planning use of a one-time script to patch
+     * payouts from historic edge cases. But it's also good to cover the data sanity check for the long term, since
+     * Stripe {@link https://docs.stripe.com/api/payouts/object#payout_object-status say} "Some payouts that fail might
+     * initially show as paid, then change to failed."
+     */
+    public function testNoOpWhenPayoutFailed(): void
+    {
+        $app = $this->getAppInstance();
+        /** @var Container $container */
+        $container = $app->getContainer();
+
+        $donation = $this->getTestDonation();
+
+        $stripeBalanceTransactionProphecy = $this->prophesize(BalanceTransactionService::class);
+        $stripeBalanceTransactionProphecy->all(
+            [
+                'limit' => 100,
+                'payout' => self::DEFAULT_PAYOUT_ID,
+            ],
+            ['stripe_account' => self::CONNECTED_ACCOUNT_ID],
+        )
+            ->shouldNotBeCalled();
```

Comment:
Can say we don't want it called with any arguments, simpler than just saying don't call it with these specific ones

```suggestion
        $stripeBalanceTransactionProphecy->all(Argument::cetera())
            ->shouldNotBeCalled();
```